.github/workflows/type_checking.yml

```diff
@@ -0,0 +1,26 @@
+name: Type checking
+
+on:
+  push:
+    paths:
+      - '*.py'
```

Comment:
won't this only trigger when the code is pushed to master?

Reply:
It should trigger when you push to a PR. (I think `paths` should actually be `'**.py'`.)

Fwiw, you can specify to only trigger on pushes to master with
```yaml
  push:
    branches: [master]
```